common/testing/taskpoller/taskpoller.go

```diff
@@ -314,7 +314,9 @@ func (p *workflowTaskPoller) respondTaskCompleted(
 	if reply.Identity == "" {
 		reply.Identity = opts.tv.WorkerIdentity()
 	}
-	reply.ReturnNewWorkflowTask = true
+	if reply.ForceCreateNewWorkflowTask {
+		reply.ReturnNewWorkflowTask = true
+	}
```

Comment:
I think the change is a bit too "magical". It sets one field true if another field is true. I'd like to avoid that.

If we cannot always set `ReturnNewWorkflowTask = true`, then I prefer we remove this entire block instead and let the user set it manually.

Reply:
@ShahabT

Reply:
My test involves concurrent activities and it becomes non deterministic if I allow wft completion to return a new wft because an activity **might or might not** have complete between the last wft start to complete. So either I need a way to disable `ReturnNewWorkflowTask` or make it disable by default.
I agree that the current code is too magical, so let's remove this block and set it true in your single test?

Reply:
Got it! That makes sense to me 👍 

In that case, let's remove it altogether.

Reply:
included the change in https://github.com/temporalio/temporal/pull/6920/files